tensorflow_datasets/image/mnist.py

```diff
@@ -111,21 +112,27 @@ def _info(self):
             "label": tfds.features.ClassLabel(num_classes=10),
         }),
         supervised_keys=("image", "label"),
-        urls=[self.URL],
+        urls=[self.URL, _MNIST_URL_FALLBACK_],
         citation=_MNIST_CITATION,
     )
 
   def _split_generators(self, dl_manager):
     """Returns SplitGenerators."""
     # Download the full MNIST Database
-    filenames = {
-        "train_data": _MNIST_TRAIN_DATA_FILENAME,
-        "train_labels": _MNIST_TRAIN_LABELS_FILENAME,
-        "test_data": _MNIST_TEST_DATA_FILENAME,
-        "test_labels": _MNIST_TEST_LABELS_FILENAME,
-    }
-    mnist_files = dl_manager.download_and_extract(
-        {k: urllib.parse.urljoin(self.URL, v) for k, v in filenames.items()})
+    def download_and_extract(url):
```

Comment:
Why create a new function?

Reply:
I think it makes try...except block easy to understand. So, let's remove the function or not?

Reply:
I think inner function is confusing so you can delete. 

You can split `dl_manager.download` and `dl_manager.extract` functions. Now try..except block is be better.

Reply:
done :-)